build_tools/aws-sdk-code-generator/spec/protocol_tests_spec_helper.rb

```diff
@@ -0,0 +1,291 @@
+# frozen_string_literal: true
+
+require 'rexml/document'
+require 'rspec/expectations'
+
+PROTOCOL_TESTS_IGNORE_LIST_PATH = File.join(File.dirname(__FILE__), 'protocol-tests-ignore-list.json')
+
+# This module contains helpers relating to protocol-tests
+module ProtocolTestsHelper
+  class << self
+
+    def check_ignore_list(protocol, test_id, test_type)
+      return nil if protocol.include?('extras')
+
+      ignore_list = JSON.parse(File.read(PROTOCOL_TESTS_IGNORE_LIST_PATH))[protocol]
```

Comment:
The efficiency probably doesn't really matter much here, but this requires us to repeatedly read and parse this file.  We should probably do it once (make a memoized method).